File: 20. Valid Parentheses/ValidParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = [];
    const mapping = {')':'(','}':'{',']':']'};
    for (let i=0; i<s.length; i++){
        if (mapping[s[i]]){
            top_stack = stack.length>0? stack.pop() : '#'
            if (mapping[s[i]] != top_element){
                return false
            }

        }
        else{
            stack.push(s[i])
        }

    }
    return stack.legth === 0;
    
};

File: 20. Valid Parentheses/ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        valid = {'(':')', '{':'}', '[':']'}
        for i in range(len(s)):
            stack.append(s[i])
            for j in range (i+1, len(s)):
                if valid[s[i]]==s[j]:
                    stack.append(s[j])
                else:
                    return False
            if len(stack)%2 == 0:
                return True
            else:
                return False
                